### PR TITLE
Make FileBasedXmlDocumentProvider not process DTDs

### LINKED ISSUE #1371: FileBasedXmlDocumentationProvider processes DTDs
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Utilities/Documentation/FileBasedXmlDocumentationProvider.cs needs to consume http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Utilities/Documentation/XmlDocumentationProvider.cs,59.

This is a security hole.

While you're at it, please also fix https://roslyn.codeplex.com/workitem/406; you can see my fixed version at https://github.com/SLaks/VSEmbed/blob/master/VSEmbed.Roslyn/XmlDocumentationProvider.cs

### PR DESCRIPTION
**Pull request by basoundr**
Fix #1371 :  Make FileBasedXmlDocumentProvider not process DTD by
setting the correct DTDProcessing value in the XmlReaderSetting

Also fix XmlDocumentProvicer to return the string with the Tags and
Newlines when called GetDocumentForSymbol

### CODE REVIEW

**Comment by basoundr** [comment]
Potential reviewers: @Pilchie @jasonmalinowski . 
Other reviewers: @dpoeschl @rchande @balajikris @brettfo

File: src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs [hunk 1]
```diff
@@ -70,13 +77,9 @@ public ContentBasedXmlDocumentationProvider(byte[] xmlDocCommentBytes)
...
 
-            protected override XDocument GetXDocument()
+            protected override Stream GetSourceStream()
             {
-                using (var stream = SerializableBytes.CreateReadableStream(_xmlDocCommentBytes, CancellationToken.None))
-                using (var xmlReader = XmlReader.Create(stream, s_xmlSettings))
-                {
-                    return XDocument.Load(xmlReader);
-                }
+                return SerializableBytes.CreateReadableStream(_xmlDocCommentBytes, CancellationToken.None);
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: Nit: it looks like we have a cancellation token coming in. Should we pass it through?

File: src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs [hunk 2]
```diff
@@ -40,7 +47,7 @@ protected override string GetDocumentationForSymbol(string documentationMemberID
                         {
                             if (e.Attribute("name") != null)
                             {
-                                _docComments[e.Attribute("name").Value] = e.Value;
+                                _docComments[e.Attribute("name").Value] = string.Concat(e.Nodes());
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: If you're bored and want to make this more efficient, just using a XmlReader directly might be more sane here. We're using an XML reader to create a full tree, only to immediately get stuff back from it. And you could call XmlReader.ReadInnerXml which would just give you the string you want directly and avoid creating a huge pile of garbage.

**Comment by jasonmalinowski** [comment]
:+1:

The perf improvements would be nice to make, but that can be tracked separately and in another commit.

File: src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs (hunk 2 above)
Comment (rchande) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: This is going to get called during typing, so it should probably try to be efficient.

**Comment by rchande** [comment]
You might be interested in this, @pharring .

**Comment by jasonmalinowski** [comment]
And for FYI, @slaks.

File: src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs (hunk 2 above)
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: In the VS case there's a separate provider. This is only used for non-VS workspaces like MSBuildWorkspace. Right @basoundr?

Comment (SLaks) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: No; VisualStudioDocumentationProvider shares this base class.

http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/DocumentationComments/VisualStudioDocumentationProvider.cs,15

Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: It has the base type of DocumentationProvider, not XmlDocumentationProvider. (I agree the names are terrible.)

Comment (basoundr) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: Yes, this change wont affect VS

File: src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs (hunk 1 above)
Comment (basoundr) [inline src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs]: @jasonmalinowski Addressed

**Pull request merged**